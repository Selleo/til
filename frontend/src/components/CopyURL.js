import React, { useState } from 'react'
import { getCurrentURL } from '../utils'
import CopyButton from './CopyButton'
import { Tooltip } from 'react-tippy'

const CopyPostURL = ({ postId }) => {
  const [isCopied, setIsCopied] = useState('Click to copy link')

  const copyURL = () => {
    let currentURL = getCurrentURL()

    if (postId) {
      currentURL = currentURL + `posts/${postId}`
    }

    navigator.clipboard.writeText(currentURL)
    setIsCopied('Copied!')

    setTimeout(() => {
      setIsCopied('Click to copy link')
    }, 300)
  }

  return (
    <Tooltip
      className="ToolTip"
      arrow
      delay={150}
      duration={500}
      html={<div>{isCopied}</div>}
    >
      <CopyButton handleClick={copyURL} text={'Share'} />
    </Tooltip>
  )
}

export default CopyPostURL
