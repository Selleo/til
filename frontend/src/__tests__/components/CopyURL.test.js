import React from "react";
import { render } from "@testing-library/react";
import CopyURL from "../../components/CopyURL";

describe("<CopyURL/>", () => {
  it("renders properly", () => {
    const { asFragment } = render(<CopyURL />);
    expect(asFragment()).toMatchSnapshot();
  });
});
