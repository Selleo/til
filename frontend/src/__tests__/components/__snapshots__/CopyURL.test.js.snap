// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CopyURL/> renders properly 1`] = `
<DocumentFragment>
  <button>
    Copy post Url
  </button>
</DocumentFragment>
`;
